distilled from a fix from upstream for c++ exception handling on sparc-fbsd

http://bugs.gentoo.org/317335
http://gcc.gnu.org/viewcvs?view=revision&revision=152027

--- gcc-4.4.3/gcc/config.gcc
+++ gcc-4.4.3/gcc/config.gcc
@@ -2300,6 +2300,10 @@
 	tm_file="${tm_file} ${fbsd_tm_file} dbxelf.h elfos.h sparc/sysv4.h sparc/freebsd.h"
 	extra_options="${extra_options} sparc/long-double-switch.opt"
 	tmake_file="${tmake_file} sparc/t-crtfm"
+	fbsd_major=`echo ${target} | sed -e 's/.*freebsd//g' | sed -e 's/\..*//g'`
+	if test ${fbsd_major} -ge 7; then
+		tmake_file="${tmake_file} t-freebsd-eh"
+	fi
 	case "x$with_cpu" in
 		xultrasparc) ;;
 		x) with_cpu=ultrasparc ;;
